### PR TITLE
Auth on redirect

### PR DESCRIPTION
**Pull request by asvetlov**
Fix for #1699

### CODE REVIEW

**Comment by asvetlov** [comment]
Committers, I need a review of the issue.
Sure, I can merge it without approval (I'm an admin btw) but I like other eyes on it

**Comment by asvetlov** [comment]
Waiting for tomorrow

File: aiohttp/client.py [hunk 1]
```diff
@@ -231,6 +222,24 @@ def _request(self, method, url, *,
...
+                    if auth is None:
+                        auth = auth_from_url
+                    if auth is None:
+                        auth = self._default_auth
+                    # It would be confusing if we support explicit
+                    # Authorization header with auth argument
+                    if (headers is not None and
+                            auth is not None and
+                            hdrs.AUTHORIZATION in headers):
+                        raise ValueError("Cannot combine AUTHORIZATION header "
```
Comment (kxepal) [inline aiohttp/client.py:239]: Can we have tests which triggers these ValueError's? I'm hardly imagine the case for them.

File: aiohttp/client.py [hunk 2]
```diff
@@ -322,6 +331,10 @@ def _request(self, method, url, *,
                         elif not scheme:
                             r_url = url.join(r_url)
 
+                        if url.origin() != r_url.origin():
```
Comment (kxepal) [inline aiohttp/client.py:334]: Side note: it seems like relative `Location` [is legit](https://tools.ietf.org/html/rfc7231#section-7.1.2) now.

File: aiohttp/client.py (hunk 1 above)
Comment (asvetlov) [reply aiohttp/client.py:239]: Yes, the PR has a test for the case

File: aiohttp/client.py (hunk 2 above)
Comment (asvetlov) [reply aiohttp/client.py:334]: Relative URL is joined to original two lines above.
`r_url` is absolute at this line.

File: aiohttp/client.py (hunk 1 above)
Comment (kxepal) [reply aiohttp/client.py:239]: Hm...seems  it's slipped away from me. Ok.

**Comment by asvetlov** [comment]
@kxepal thank you for careful review.
Relative urls was processed by aiohttp as I've commented.

File: aiohttp/client.py (hunk 2 above)
Comment (kxepal) [reply aiohttp/client.py:334]: I'm interpreted it as schemaless url rather then relative one. Fine then.

**Pull request merged**